cirq-core/cirq/transformers/noise_adding_test.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cirq import ops, circuits, devices
+import cirq.transformers.noise_adding as na
+
+
+def test_noise_adding():
+    qubits = devices.LineQubit.range(4)
+    circuit = circuits.Circuit(ops.CZ(*qubits[:2]), ops.CZ(*qubits[2:])) * 10
+    transformed_circuit_p0 = na.DepolerizingNoiseTransformer(0.0)(circuit)
+    assert transformed_circuit_p0 == circuit
+    transformed_circuit_p1 = na.DepolerizingNoiseTransformer(1.0)(circuit)
+    assert len(transformed_circuit_p1) == 20
+    transformed_circuit_p0_03 = na.DepolerizingNoiseTransformer(0.03)(circuit)
+    assert 10 <= len(transformed_circuit_p0_03) <= 20
```

Comment:
isn't the expected len of the circuit num_moment * (1 + p) ? so here we expect ~$10 * 1.03 = 13$ and 20 correspends to the extreme $p = 1$ ... I think these tests are bad tests ... instead of doing this randomly call the transformer with a specific generator so that the result is always the same circuit and check equality with that circuit